org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java

```diff
@@ -0,0 +1,184 @@
+
+				for (final WorkParameterEvolution workParam : usage.getWorkEvolutions()) {
+					final VariableCharacterisation varChar = workParam.getVariableCharacterisation();
+					final Sequence paramSequence = workParam.getEvolution();
+					if (varChar == null) {
+						LOGGER.error("Skipping evolution of unspecified work parameter");
+						continue;
+					}
+					if (paramSequence == null) {
+						LOGGER.error("Skipping unspecified evolution for work parameter " + varChar);
+						continue;
+					}
+
+					// Add parameter and model evaluator for the work parameter
+					workEvaluators.put(varChar, new ModelEvaluator(paramSequence));
+				}
+			}
+
+	}
+
+	/**
+	 * Actually evolves the load.
+	 */
+	public void triggerInternal(final double time) {
+
+		loadEvaluator.ifPresent(evaluator -> this.evolveLoad(evaluator, time));
+
+		// Then, iterate through work parameters to evolve
+		for (final VariableCharacterisation workParam : this.workEvaluators.keySet()) {
+			this.evolveWork(workParam, workEvaluators.get(workParam), time);
+		}
+	}
+
+	/**
+	 * The length of <code>this</code>' DLIM sequence.
+	 *
+	 * @return The length of <code>this</code>' DLIM sequence.
+	 */
+	protected double getDLIMFinalDuration() {
+		return this.usage.getLoadEvolution().getFinalDuration();
+	}
+
+	/**
+	 * Evolve the load.
+	 *
+	 * @param loadEvaluator DLIM evaluator used to fetch the load at the current
+	 *                      point in time.
+	 */
+	protected void evolveLoad(final ModelEvaluator loadEvaluator, final double time) {
+
+		double newRate = this.getNewRate(loadEvaluator, time);
+		final Workload wl = this.usage.getScenario().getWorkload_UsageScenario();
+		if (wl != null) {
+			if (wl instanceof OpenWorkload) {
+				final PCMRandomVariable openwl = ((OpenWorkload) wl).getInterArrivalTime_OpenWorkload();
+				if (newRate != 0) {
+					// Using inverse value to convert from arrival rate to inter arrival
+					// time
+					newRate = 1 / newRate;
+				} else {
+					// If the arrival rate is 0, the inter arrival time should be
+					// infinite. We use the max integer value as an approximation
+					newRate = Integer.MAX_VALUE;
+				}
+
+				final String newRateStr = Double.toString(newRate);
+				if (newRateStr.equals(openwl.getSpecification())) {
+					LOGGER.debug("Inter arrival time is still: " + newRateStr);
+				} else {
+					LOGGER.debug(
+							"Changing inter arrival time from: " + openwl.getSpecification() + " to :" + newRateStr);
+					openwl.setSpecification(newRateStr);
+				}
+			} else if (wl instanceof ClosedWorkload) {
+				final int newRateInt = (int) Math.round(newRate);
+				final int oldRate = ((ClosedWorkload) wl).getPopulation();
+				if (newRateInt == oldRate) {
+					LOGGER.debug("Closed workload population is still: " + newRateInt);
+				} else {
+					LOGGER.debug("Changing closed workload population from: " + oldRate + " to " + newRateInt);
+					((ClosedWorkload) wl).setPopulation(newRateInt);
```

Comment:
This does not affect the simulation at the moment. The users are spawned on Simulation Start.

Reply:
True. I cannt do anything about that though.

Reply:
Maybe throwing, in addition, a ModelChanged event for both, while here the implementation assumes that somebody will pick it up correctly. It's like in SPD; when the interpreter changes the model by transformations, it throws events. Then it should be up to the other extensions to say look, if this part changes, then indeed I need to do something or state hey, I am stateless. I don't care about this change.